### PR TITLE
Fix a crash consistency bug in MDC open

### PR DESCRIPTION
**Pull request by nabeelmmd**
A crash during MDC erase could leave the new passive log header partially updated. On MDC reopen, we detect this crash situation correctly but the passive log handle is erroneously freed causing MDC open to fail with EINVAL.

The fix is to not free the passive log handle on detecting this crash situation so that the subsequent erase of the passive log and the mdc open succeeds.

Signed-off-by: Nabeel M Mohamed <50154757+nabeelmmd@users.noreply.github.com>
Reported-by: Yile (Michael) Gu

### CODE REVIEW

**Review by tristan957 — APPROVED** [review]

**Comment by tristan957** [comment]
Nabeel in your commit message/PR, can you add `Fixes #471`?

**Comment by nabeelmmd** [comment]
> Nabeel in your commit message/PR, can you add `Fixes #471` prior to the commit trailers? This will auto close the issue when the PR is merged

Nice! I didn't know about this.
https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

**Comment by tristan957** [comment]
It is a nice feature. We should move to GitHub issues instead of JIRA to take advantage of this! 😄

File: lib/mpool/lib/mdc_file.c [hunk 1]
```diff
@@ -396,25 +394,26 @@ mdc_file_open(
...
 
+    /* On detecting an incomplete log header from a crash during mdc file erase,
+     * return a valid handle.
+     */
     err = mdc_file_validate(mfp, gclose, gen);
-    if (err) {
+    if (err && merr_errno(err) != ENOMSG) {
         mdc_file_unmap(mfp);
-        goto err_exit1;
+        goto err_exit;
```
Comment (beckerg) [inline lib/mpool/lib/mdc_file.c:405]: Maybe comment should say something about returning ENOMSG as well as a valid handle...

**Review by beckerg — APPROVED** [review]

**Pull request merged**